pageserver/src/tenant.rs

```diff
@@ -120,6 +126,10 @@ pub struct Tenant {
 
     /// Makes every timeline to backup their files to remote storage.
     upload_layers: bool,
+
+    /// Cached logical sizes updated updated on each [`Tenant::collect_pricing_operations`].
+    // TODO: Lsn: Hash
+    cached_logical_sizes: tokio::sync::Mutex<HashMap<(TimelineId, u64), u64>>,
```

Comment:
I don't see why Lsn shouldn't implement Hash, but didn't yet search any archives for a reason.

Reply:
Derived this.